server/athenian/api/internal/miners/filters.py

```diff
@@ -116,6 +125,36 @@ def __bool__(self) -> bool:
             ],
         )
 
+    def __or__(self, other: JIRAFilter) -> JIRAFilter:
+        """Return a new JIRAFilter which is logical union of this filter with another."""
+        if (
+            ((self.account != other.account) and self.account and other.account)
+            or self.unmapped
+            or isinstance(self.epics, bool) != isinstance(other.epics, bool)
+        ):
+            raise ValueError("Cannot union JIRAFilter with different accounts, unmapped or epics")
+
+        account = self.account if self.account and other.account else 0
+
+        if isinstance(self.epics, bool):
+            epics: bool | frozenset[str] = False
+        else:
+            assert not isinstance(other.epics, bool)
+            epics = _join_filter_sets(self.epics, other.epics)
+
+        projects = tuple(_join_filter_sets(frozenset(self.projects), frozenset(other.projects)))
```

Comment:
Why not define `projects` as frozenset in the class? We don't cast it to numpy arrays, most likely.